Objects/floatobject.c

```diff
@@ -1683,8 +1684,97 @@ typedef enum _py_float_format_type float_format_type;
 #define ieee_big_endian_format _py_float_format_ieee_big_endian
 #define ieee_little_endian_format _py_float_format_ieee_little_endian
 
-#define float_format (_PyRuntime.float_state.float_format)
-#define double_format (_PyRuntime.float_state.double_format)
+#ifdef __GNUC__
+__attribute__((always_inline))
+#endif
+static inline uint32_t
+byte_swap_uint32(uint32_t x) {
+    return (
+        ((x >> 24) & 0xff)
+        | ((x >> 8) & 0xff00)
+        | ((x & 0xff00) << 8)
+        | ((x & 0xff) << 24)
+    );
+}
+
+#ifdef __GNUC__
+__attribute__((always_inline))
+#endif
+static inline uint64_t
+byte_swap_uint64(uint64_t x) {
+    return (
+        ((x >> 56) & 0xff)
+        | ((x >> 40) & 0xff00)
+        | ((x >> 24) & 0xff0000)
+        | ((x >> 8) & 0xff000000)
+        | ((x & 0xff000000) << 8)
+        | ((x & 0xff0000) << 24)
+        | ((x & 0xff00) << 40)
+        | ((x & 0xff) << 56)
+    );
+}
+
+/* We attempt to determine if this machine is using IEEE
+    floating point formats by peering at the bits of some
+    carefully chosen values.  If it looks like we are on an
+    IEEE platform, the float packing/unpacking routines can
+    just copy bits, if not they resort to arithmetic & shifts
+    and masks.  The shifts & masks approach works on all finite
+    values, but what happens to infinities, NaNs and signed
+    zeroes on packing is an accident, and attempting to unpack
+    a NaN or an infinity will raise an exception.
+
+    Note that if we're on some whacked-out platform which uses
+    IEEE formats but isn't strictly little-endian or big-
+    endian, we will fall back to the portable shifts & masks
+    method.
+
+    These functions are reduced to a no-op via compiler
+    optimizations. */
+
+#ifdef __GNUC__
+__attribute__((always_inline))
+#endif
+static inline float_format_type
+get_float_format(void) {
+    if (sizeof(float) == 4) {
```

Comment:
Why did you extract this function? I think 1) it's better if we keep an inline version (in the end you want it to be inline, so you can, create a macro also).